scripts/utils/price-utils.js

```diff
@@ -139,9 +139,52 @@ module.exports = function(web3 = web3, artifacts = artifacts) {
     return true
   }
 
+  const checkNoProfitableOffer = async (order, exchange, currentMarketPrice) => {
+    const tokenInfo = fetchTokenInfoFromExchange(exchange, [order.buyToken, order.sellToken])
+
+    if (currentMarketPrice === undefined) {
+      currentMarketPrice = await getDexagPrice(
+        (await tokenInfo[order.buyToken]).symbol,
+        (await tokenInfo[order.sellToken]).symbol
+      )
+    }
+    // checks whether the order amount is negligible
+    if ((await valueInUSD(order, tokenInfo)) < 10) {
+      return true
+    }
+
+    // multiplicator is used to counter rounding errors
+    const multiplicator = 1000000000
+    const marketPrice = new BN(currentMarketPrice * multiplicator)
+    const orderPrice = order.priceNumerator
+      .mul(new BN(10).pow(new BN((await tokenInfo[order.sellToken]).decimals)))
+      .div(new BN(10).pow(new BN((await tokenInfo[order.buyToken]).decimals)))
+      .mul(new BN(multiplicator))
+      .div(order.priceDenominator)
+
+    return marketPrice.lt(orderPrice)
```

Comment:
Since computing prices is usually a source of mistakes, I wonder if it makes sense to create a standalone function converting prices to some version of "BN float" like you are doing here. Then this function could be tested separately and then we could rely on that everywhere.

Reply:
I think one can just use your existing function: `toErc20Units`. Please double check

Reply:
Mmh, I'm not sure how you propose to use `toErc20Units` here. My proposal would be to import the `Fraction` class that Felix created and extend it with a `fromNumber` method that creates a fraction approximation of a float (and for practicality also extend it with `.mulBN`, `.divBN`). Then we can use the code you wrote above, even without the multiplicator.